src/main/java/bio/overture/ego/repository/queryspecification/AbstractPermissionSpecification.java

```diff
@@ -0,0 +1,74 @@
+package bio.overture.ego.repository.queryspecification;
+
+import static bio.overture.ego.model.entity.AbstractPermission.Fields.accessLevel;
+import static bio.overture.ego.model.entity.AbstractPermission.Fields.policy;
+import static bio.overture.ego.model.enums.JavaFields.*;
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+import bio.overture.ego.model.entity.AbstractPermission;
+import bio.overture.ego.model.entity.Policy;
+import bio.overture.ego.model.search.SearchFilter;
+import bio.overture.ego.utils.QueryUtils;
+import com.google.common.collect.Lists;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Stream;
+import javax.persistence.criteria.Predicate;
+import lombok.NonNull;
+import lombok.val;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.security.core.parameters.P;
+
+public abstract class AbstractPermissionSpecification extends SpecificationBase {
```

Comment:
although this seems intuitive, you cannot refactor these Type bounded generics this way. <P> is not defined and so means nothing. Please replicate the logic before as was done in
https://github.com/overture-stack/ego/commit/5a5ce1e19847ba6204c06adbdf0947dd66cd0068#diff-29b6a77729ad77239fcf9b4a22d26536